nsxt/resource_nsxt_policy_security_policy_test.go

```diff
@@ -240,35 +350,64 @@ func testAccNsxtPolicySecurityPolicyExists(resourceName string, domainName strin
 			return fmt.Errorf("Policy SecurityPolicy resource ID not set in resources")
 		}
 
-		_, err := nsxClient.Get(domainName, resourceID)
-		if err != nil {
-			return fmt.Errorf("Error while retrieving policy SecurityPolicy ID %s. Error: %v", resourceID, err)
+		isPolicyGlobalManager := isPolicyGlobalManager(testAccProvider.Meta())
+		if !resourceNsxtPolicySecurityPolicyExistsInDomain(resourceID, domainName, connector, isPolicyGlobalManager) {
+			return fmt.Errorf("Error while retrieving policy SecurityPolicy ID %s", resourceID)
 		}
-
 		return nil
 	}
 }
 
 func testAccNsxtPolicySecurityPolicyCheckDestroy(state *terraform.State, displayName string, domainName string) error {
 	connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
-	nsxClient := domains.NewDefaultSecurityPoliciesClient(connector)
+	var nsxClient interface{}
+	isPolicyGlobalManager := isPolicyGlobalManager(testAccProvider.Meta())
+	if isPolicyGlobalManager {
+		nsxClient = gm_domains.NewDefaultSecurityPoliciesClient(connector)
+	} else {
+		nsxClient = domains.NewDefaultSecurityPoliciesClient(connector)
+	}
+	var err error
 	for _, rs := range state.RootModule().Resources {
 
 		if rs.Type != "nsxt_policy_security_policy" {
 			continue
 		}
 
 		resourceID := rs.Primary.Attributes["id"]
-		_, err := nsxClient.Get(domainName, resourceID)
+		if isPolicyGlobalManager {
+			_, err = nsxClient.(*gm_domains.DefaultSecurityPoliciesClient).Get(domainName, resourceID)
```

Comment:
I think we can use resourceNsxtPolicySecurityPolicyExistsInDomain here as well?

Reply:
Thanks for the catch! fixed.